javascript/src/components/streams/StreamComponent.jsx

```diff
@@ -0,0 +1,81 @@
+'use strict';
+
+var React = require('react/addons');
+var StreamsStore = require('../../stores/streams/StreamsStore');
+var StreamList = require('./StreamList');
+var CreateStreamButton = require('./CreateStreamButton');
+var SupportLink = require('../support/SupportLink');
+var PermissionsMixin = require('../../util/PermissionsMixin');
+var StreamRulesStore = require('../../stores/streams/StreamRulesStore');
+
+var StreamComponent = React.createClass({
+    mixins: [PermissionsMixin],
+    getInitialState() {
+        return {
+            streams: []
+        };
+    },
+    componentDidMount() {
+        this.loadData();
+        StreamsStore.onChange(this.loadData);
+        StreamRulesStore.onChange(this.loadData);
```

Comment:
Then I would load the list of stream rules types once, when the component mount, and only reload the list of streams when something changes. I know it's a really cheap rest call, but it's always nice saving some of those.